pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -811,9 +826,13 @@ func (gce *GCECloud) targetPoolNeedsUpdate(name, region string, affinityType api
 		if isHTTPErrorCode(err, http.StatusNotFound) {
 			return false, true, nil
 		}
-		return false, false, fmt.Errorf("error getting load balancer's target pool: %v", err)
+		// Err on the side of caution in case of errors. Caller should notice the error and retry.
+		// We never want to end up recreating resources because gce api flaked.
+		return true, false, fmt.Errorf("error getting load balancer's target pool: %v", err)
 	}
-	if translateAffinityType(affinityType) != tp.SessionAffinity {
+	// TODO: This is fickle, if either side changes defaults it could backfire.
+	if tp.SessionAffinity != "" && translateAffinityType(affinityType) != tp.SessionAffinity {
```

Comment:
safety first? this will just prevent the resync for session affinity. or would you prefer treating empty as none?